snap/snapcraft.yaml

```diff
@@ -5,12 +5,69 @@ license: BSD-3-Clause
-summary: DERP Relay Server for Tailscale
+issues: https://github.com/canonical/derper-snap/issues
+source-code: https://github.com/canonical/derper-snap
+summary: DERP Server for Tailscale
 description: |
-  Self hosted DERP Relay servers for Tailscale networks.
-  This snap provides a simple way to run your own DERP servers to route your tailscale traffic through your own relay servers.
+  Derper is a self hosted DERP (Designated Encrypted Relay for Packets) server, designed to be used with Tailscale.
+  This snap provides a simple and secure way to run your own DERP servers to route your tailscale traffic through your own relay servers.
+
+  Please refer to the upstream documentation at https://tailscale.com/kb/1118/custom-derp-servers and https://tailscale.com/kb/1232/derp-servers for more information.
+
+  **Usage**
+
+  Install the snap:
+
+     sudo snap install derper
+
+  This installs the DERP server and starts it automatically.
+  You will likely need to configure the service:
+
+  **Configuration**
+
+  You can modify configuration options with `sudo snap set`. The following configuration options are currently supported:
+
+  * `hostname`: LetsEncrypt host name, if addr's (`a`) port is :443. (default if unset: "derp.tailscale.com")
+  * `a`: server HTTP/HTTPS listen address, in form ":port", "ip:port", or for IPv6 "[ip]:port". If the IP is omitted, it defaults to all interfaces. Serves HTTPS if the port is 443 and/or certmode is manual (default certmode is "letsencrypt"), otherwise HTTP. (default if unset: ":443")
+  * `stun-port`: The UDP port on which to serve STUN. The listener is bound to the same IP (if any) as specified in the `a` option. (default if unset: "3478")
+
+  The config options correspond to the similarly named command line flags for `derper`.
+
+  Options that don't have support to be configured through the snap will be left on their default.
+  See the `derper` command source code at https://github.com/tailscale/tailscale/tree/main/cmd/derper for more technical information on options.
+
+  Example of setting an option:
+
+     sudo snap set derper hostname=myderper.mydomain.com
+
+  To view the available config options and their current values:
+
+     $ sudo snap get derper
+     Key             Value
+     hostname
+     a               127.0.0.1:443
+     ...
+
+  Derper must be restarted manually for the changed config to take affect:
+
+     sudo snap restart derper
+
+  **Security**
+
+  Derper provides a way to negotiate direct connections between devices running tailscale,
+  or acting as a full relay server if no direct connection is possible.
+  Derper never sees the content of traffic it relays, as all data is end to end encrypted already by Tailscale using Wireguard on the end devices.
+  Please see the upstream documentation for more security information about derper.
+
+  The snap is strictly confined, with interfaces limited to:
+
+  * `network`: general network access.
```

Comment:
I am not sure if you can use `*` for the list instead of `-`, so I hope you tested it on Snapcraft

Reply:
Yep I've been following the markdown syntax guide on the snap store listing page:

![1733262812](https://github.com/user-attachments/assets/f71861b0-999f-4e5b-9e8e-a158493117ac)

Reply:
The **Code** section is not quite accurate though - see https://github.com/canonical/snapcraft.io/issues/4611